image_view/src/nodes/extract_images.cpp

```diff
@@ -34,84 +34,90 @@
 
 #include <opencv2/highgui/highgui.hpp>
 
-#include <ros/ros.h>
-#include <sensor_msgs/Image.h>
+#include <rclcpp/rclcpp.hpp>
+#include <sensor_msgs/msg/image.h>
 #include <cv_bridge/cv_bridge.h>
 #include <image_transport/image_transport.h>
 
-#include <boost/thread.hpp>
+#include <thread>
+#include <memory>
+#include <string>
 #include <boost/format.hpp>
 
+using std::placeholders::_1;
 class ExtractImages
 {
 private:
+  rclcpp::Node::SharedPtr node;
   image_transport::Subscriber sub_;
-
-  sensor_msgs::ImageConstPtr last_msg_;
-  boost::mutex image_mutex_;
-
+  sensor_msgs::msg::Image::ConstSharedPtr last_msg_;
+  std::mutex image_mutex_;
+  
   std::string window_name_;
   boost::format filename_format_;
   int count_;
-  double _time;
-  double sec_per_frame_;
+  rclcpp::Time _time;
+  rclcpp::Time sec_per_frame_;
 
 #if defined(_VIDEO)
   CvVideoWriter* video_writer;
 #endif //_VIDEO
 
 public:
-  ExtractImages(const ros::NodeHandle& nh, const std::string& transport)
-    : filename_format_(""), count_(0), _time(ros::Time::now().toSec())
+  ExtractImages(const std::string& topic,
+                const std::string& transport)
+    : filename_format_(""), count_(0), _time(rclcpp::Clock().now())
   {
-    std::string topic = nh.resolveName("image");
-    ros::NodeHandle local_nh("~");
-
+    // auto clock_ = rclcpp::Clock();
+    node = rclcpp::Node::make_shared("extract_images");
     std::string format_string;
-    local_nh.param("filename_format", format_string, std::string("frame%04i.jpg"));
+    node->get_parameter_or("filename_format", format_string, std::string("frame%04i.jpg"));
     filename_format_.parse(format_string);
-
-    local_nh.param("sec_per_frame", sec_per_frame_, 0.1);
-
-    image_transport::ImageTransport it(nh);
-    sub_ = it.subscribe(topic, 1, &ExtractImages::image_cb, this, transport);
+    double sec_per_frame;
+    node->get_parameter_or("sec_per_frame", sec_per_frame, 0.1);
+    sec_per_frame_ = rclcpp::Time(sec_per_frame);
+    std::shared_ptr<image_transport::ImageTransport> it;
+    it.reset(new image_transport::ImageTransport(node));
+    image_transport::TransportHints hints(node, transport);
+    sub_ = it->subscribe(topic, 1, std::bind(&ExtractImages::image_cb, this, _1), node, &hints);
```

Comment:
please use new pointer api and ref to https://github.com/ros-perception/image_pipeline/blob/843b932985acc41889071b1749a1e13d80ea58c2/depth_image_proc/src/nodelets/crop_foremost.cpp#L93